modules/rank-eval/src/main/java/org/elasticsearch/index/rankeval/RankEvalResult.java

```diff
@@ -31,43 +31,49 @@
  * For each precision at n computation the id of the search request specification used to generate search requests is returned
  * for reference. In addition the averaged precision and the ids of all documents returned but not found annotated is returned.
  * */
+// TODO do we need an extra class for this or it RankEvalResponse enough?
 public class RankEvalResult implements Writeable {
-    /**ID of specification this result was generated for.*/
-    private int specId;
-    /**Average precision observed when issueing query intents with this spec.*/
+    /**ID of QA specification this result was generated for.*/
+    private String specId;
+    /**Average precision observed when issuing query intents with this specification.*/
     private double qualityLevel;
     /**Mapping from intent id to all documents seen for this intent that were not annotated.*/
-    private Map<Integer, Collection<String>> unknownDocs;
+    private Map<String, Collection<String>> unknownDocs;
```

Comment:
I was thinking if it maybe makes sense to not only return the unknown docs per query in the original request but also include partial quality metrics per query. Maybe its interesting to see e.g. which of the N original queries in the whole qa request performed bad compared to the others. Not sure though, so this is maybe a topic for further discussion.